src/test/fuzz/util.cpp

```diff
@@ -524,7 +525,23 @@ CNetAddr ConsumeNetAddr(FuzzedDataProvider& fuzzed_data_provider) noexcept
     } else if (network == Network::NET_INTERNAL) {
         net_addr.SetInternal(fuzzed_data_provider.ConsumeBytesAsString(32));
     } else if (network == Network::NET_ONION) {
-        net_addr.SetSpecial(fuzzed_data_provider.ConsumeBytesAsString(32));
+        auto pub_key{fuzzed_data_provider.ConsumeBytes<uint8_t>(torv3::ADDR_SIZE)};
+        pub_key.resize(torv3::ADDR_SIZE);
+        // Create valid onion address to prevent the address checksum from
+        // being a fuzz blocker.
+        uint8_t check_sum[torv3::CHECKSUM_LEN];
+        torv3::Checksum({pub_key.data(), pub_key.size()}, check_sum);
+
+        std::vector<uint8_t> onion_addr{pub_key};
+        onion_addr.push_back(check_sum[0]);
+        onion_addr.push_back(check_sum[1]);
+        onion_addr.push_back(torv3::VERSION[0]);
+        assert(onion_addr.size() == torv3::TOTAL_LEN);
+
+        auto encoded{EncodeBase32({onion_addr.data(), onion_addr.size()})};
+        encoded += ".onion";
+        const bool ok{net_addr.SetSpecial(encoded)};
+        assert(ok);
```

Comment:
wouldn't it be interesting `assert` that the net_addr is valid (`IsValid`)?

Reply:
I don't think so. Looking at `CNetAddr::IsValid`, there are no validity checks regarding Tor addresses, so even if a Tor address is malformed `IsValid` would return `true`.